src/licensedcode/plugin_license.py

```diff
@@ -116,3 +117,56 @@ def get_scanner(
             license_text_diagnostics=license_text_diagnostics,
             license_url_template=license_url_template
         )
+
+    def process_codebase(self, codebase, **kwargs):
+        
+        if codebase.has_single_resource:
+            return
+
+        for resource in codebase.walk(topdown=False):
+            match_reference_license(resource,codebase)
+
+
+def match_reference_license(resource, codebase):
+    """
+    Return the updated license matches having reference to another files
+    """
+    if not resource.is_file:
+        return
+
+    licenses = resource.licenses
+    license_expressions = resource.license_expressions
+    if not licenses:
+        return 
+
+    referenced_licenses = []
+    referenced_license_expressions = []
+    unique_files = []
+
+    for license in licenses:
+        referenced_filenames = license['matched_rule']['referenced_filenames']
+        for referenced_filename in referenced_filenames:
+            if referenced_filename.startswith('usr/share/common-licenses') or referenced_filename in unique_files:
+                continue
+            unique_files.append(referenced_filename)
+            new_resource = find_reference_licenses(referenced_filename=referenced_filename, resource=resource, codebase=codebase)
+            if new_resource:
+                referenced_licenses.extend(new_resource.licenses)
+                referenced_license_expressions.extend(new_resource.license_expressions)
+
+    licenses.extend(referenced_licenses)
+    license_expressions.extend(referenced_license_expressions)
+    codebase.save_resource(resource)
+    return resource
+
+
+def find_reference_licenses(referenced_filename, resource, codebase, **kwargs):
+    """
+    Return the resource object for matching referenced-filename given a resource in codebase
+    """
+    parent = resource.parent(codebase)
+    
+    for child in parent.children(codebase):
+        path = child.path
+        if path.endswith(referenced_filename):
```

Comment:
This `endswith` makes this rigid wrt to the file extension, which maybe we should not do. 

For example if a Rule has only referenced a file called `COPYING` and the actual file is `COPYING.rst` or `COPYING.md` we still should be able to find this file. This is also the case if we have a rule with `see license in LICENSE` which will match even if there are extensions, in absence of more specific rules with those extensions.

Often `LICENSE` is in the extension and not the name, that should also be followed through correctly.

All these cases should be a unit test, for this function. Similar to [this comment](https://github.com/nexB/scancode-toolkit/pull/2616/files#r690223997).

There should also be a test then of two files with the same name being present, say there are two files with `LICENSE` in it's name `apache-2.0.LICENSE` is present in the same directory, in addition to a `bsd-new.LICENSE` file being present, then this function would just return on the first License it encounters, where we should ideally add to the license-expression both. This should also be a unit test.

@pombredanne does these make sense?

Reply:
OK! I have now added a check for checking base name too